### ISSUE TITLE
Travis MacOS tests print too much

### ISSUE THREAD

**Issue by richardliaw** [ray-project/ray#1465]
### Describe the problem
Mac tests often fail due to logs being too long (and probably don't need to be):

```
The log length has exceeded the limit of 4 MB (this usually means that the test suite is raising the same exception over and over).

The job has been terminated.
```

### Source code / logs
https://api.travis-ci.org/v3/job/331567663/log.txt

**Comment by robertnishihara** [comment]
One solution would be to switch to `pytest` (at least all new tests should use `pytest`). These are much much less verbose (basically they only print stdout/stderr if a test fails).

**Comment by richardliaw** [comment]
that looks solid - I'll convert some rllib/tune tests.

**Comment by richardliaw** [comment]
addressed in #1478
